testing/test_doctest.py

```diff
@@ -129,6 +129,52 @@ def test_multiple_patterns(self, testdir):
             '*1 passed*',
         ])
 
+    def test_encoding_ascii(self, testdir):
```

Comment:
I think you could use `@pytest.mark.parametrize` on the tests to avoid repeating yourself. Something like (already incorporating my request to change to an `ini` option):

```python
import pytest

@pytest.mark.parametrize('string, encoding', [
    (u'foo', 'ascii'),
    (u'üäö', 'latin1'),
    (u'üäö', 'utf-8'),
])
def test_encoding_latin1(testdir, string, encoding):
    """Test support for doctest_encoding option.
    """
    testdir.makeini("""
        [pytest]
        doctest_encoding={0}
    """.format(encoding))
    testdir._makefile(".txt", [u"""
        >>> len(u'{0}')
        3
    """.format(string)], {})#, encoding=encoding)

    result = testdir.runpytest()
    result.stdout.fnmatch_lines([
        '*1 passed*',
    ])
```